### DISCUSSION TITLE
Change Postgres User's Password?

### DISCUSSION THREAD

**Discussion by MAXimized490** [requarks/wiki#6974] [Help / Questions] [answered]
I've been running an instance of wiki.js via docker-compose and everything has been great. I was checking some things and noticed that I never changed the default db password. It's my understanding that this is not as simple as changing the compose file because that variable is only used on initial build of the database, and so I would need to manually update the postgres user's password on the cli.

Fair enough, but I'm having a lot of trouble doing that. I'm reaching out more as a sanity check to see if what I'm trying to do is even worth it or if I should just nuke my whole install, set it up again, and import from the GitHub backup I have configured.

More details:
I am using Portainer for my Docker management but I've also tried straight from the CLI on my Ubuntu Server host. Based on the docker-compose file, I'd expect there to be a postgres user named `wiki` and there'd be a database called `wikijs`. Using `psql`, `psql -U wiki`, and `psql -U wikijs` does not work. Using sudo results in command not found, which is a good thing, but maybe I need it for this to work? Is there another way to go about switching the db user's password?

**Comment by fnafox** [comment, marked as answer]
Start by checking the environment in your wiki container as that will give you the necessary variables.
If your wiki container is called wiki then use: `docker exec -it wiki env`

Use that information to connect to the database inside the postgres container: `psql -U wikijs wiki`
When connected to the postgresql prompt, use `\password` to change the user's password.
(I'm actually not sure if this is at all necessary when using compose or if just changing the compose file is enough)

At that point the db connection for your wiki will fail, so you need to change the variable in your compose file and restart the containers. `docker-compose up -d --force-recreate` or the appropriate options in portainer.

**Reply by MAXimized490** [reply]
Ah, silly me! I had the wrong syntax for the psql command. I was just using `psql -U wikijs` or `psql -U wiki`. Thanks for the clarification! Everything is working (and secure) now!

**Reply by mittkkoo** [reply]
Is it necessary to change the db password if inside a container? I remember reading somewhere in the docs or in some issue not to change the password.
